<?xml version="1.0"?>
<launch>

    <arg name="rviz" default="true" />
    <arg name="project" default="multi_sensor_mapping"/>

    <!-- Lidar IMU Mapper -->
    <node pkg="$(arg project)" type="surfel_map_generator" name="surfel_map_generator" output="screen" respawn="false">
        <param name="session_path" value="/home/hkw/DATA/Dataset/MappingBackpackV3/公元大厦/GongyuanBuildingParkingLot(20230317)-Refined-20230324"/>
        <param name="leaf_resolution" value="0.1"/>
        <param name="depth_level" value="8"/>
        <param name="query_min_depth" value="1"/>
        <param name="qurery_max_depth" value="5"/>
        <param name="min_planarity" value="0.5"/>
    </node>


    <!--- Rviz -->
    <group if="$(arg rviz)">
      <node pkg="rviz" type="rviz" name="$(arg project)_rviz" args="-d $(find multi_sensor_mapping)/rviz/lidar_mapper.rviz" />
    </group>
</launch>
